clang/lib/Driver/ToolChains/HLSL.cpp

```diff
@@ -122,6 +126,30 @@ std::string tryParseProfile(StringRef Profile) {
     return "";
 }
 
+bool isLegalValidatorVersion(StringRef ValVersionStr, std::string &ErrorMsg) {
+  auto VerPair = ValVersionStr.split(".");
```

Comment:
You should use VersionTuple::tryParse here instead of splitting and manually parsing. That will also simplify the code below.